test/FunctionalTests/Microsoft.OData.Core.Tests/Json/JsonWriterBaseTests.cs

```diff
@@ -601,6 +602,451 @@ public void WriteMixedJsonWithLargeValues()
+                var st = jsonWriter.StartStreamValueScope();
+                WriteByteArrayInChunks(st, input);
+                jsonWriter.EndStreamValueScope();
+                jsonWriter.Flush();
+
+                stream.Seek(0, SeekOrigin.Begin);
+
+                string expected = Convert.ToBase64String(input);
+
+                using (StreamReader reader = new StreamReader(stream, encoding: Encoding.UTF8))
+                {
+                    string rawOutput = reader.ReadToEnd();
+                    Assert.Equal($"\"{expected}\"", rawOutput);
+                }
+            }
+        }
+
+        [Theory]
+        [InlineData("text/plain")]
+        [InlineData("text/html")]
+        public void CorrectlyStreamsLargeStringsToOutput(string contentType)
+        {
+            int inputLength = 1024 * 1024; // 1MB
+            string input = new string('a', inputLength);
+
+            using (MemoryStream stream = new MemoryStream())
+            {
+                IJsonWriter jsonWriter = CreateJsonWriter(stream, false, Encoding.UTF8);
+
+                var tw = jsonWriter.StartTextWriterValueScope(contentType);
+
+                WriteLargeStringInChunks(tw, input);
+
+                jsonWriter.EndTextWriterValueScope();
+                jsonWriter.Flush();
+
+                stream.Seek(0, SeekOrigin.Begin);
+
+                using (StreamReader reader = new StreamReader(stream, encoding: Encoding.UTF8))
+                {
+                    string rawOutput = reader.ReadToEnd();
+                    Assert.Equal($"\"{input}\"", rawOutput);
+                }
+            }
+        }
+
+        [Fact]
+        public void CorrectlyStreamsLargeStringsToOutput_ApplicationJson_ContentType()
+        {
+            int inputLength = 1024 * 1024; // 1MB
+            string input = new string('a', inputLength);
+
+            using (MemoryStream stream = new MemoryStream())
+            {
+                IJsonWriter jsonWriter = CreateJsonWriter(stream, false, Encoding.UTF8);
+
+                var tw = jsonWriter.StartTextWriterValueScope("");
+
+                WriteLargeStringInChunks(tw, input);
+
+                jsonWriter.EndTextWriterValueScope();
+                jsonWriter.Flush();
+
+                stream.Seek(0, SeekOrigin.Begin);
+
+                using (StreamReader reader = new StreamReader(stream, encoding: Encoding.UTF8))
+                {
+                    string rawOutput = reader.ReadToEnd();
+                    Assert.Equal(input, rawOutput);
+                }
+            }
+        }
+
+        [Theory]
+        [InlineData("text/plain")]
+        [InlineData("text/html")]
+        public void CorrectlyStreamsLargeStringsWithSpecialCharactersToOutput(string contentType)
+        {
+            int inputLength = 1024 * 1024; // 1MB
+            string input = new string('a', inputLength);
+
+            // Append special characters to the input string
+            input += "!@#$%^&*()";

```

Comment:
For special characters, I suggest using characters that need to be escaped as well as characters that take more than one char or multiple bytes to encode. Characters that many need escaping include new lines, double quotes and other control characters. Characters that require multiple bytes or multiple `char` to represent are those unicode characters outside the ASCII range, for example emojis or characters from non latin alphabets or scripts. See: https://learn.microsoft.com/en-us/dotnet/standard/base-types/character-encoding-introduction

Reply:
I also think it would be better to spread the special chars across the input instead of just at the end. More specifically, we should make sure have scenarios the verify that characters which are spread across buffer or chunksize boundaries are written correctly.

Reply:
For example an emoji that takes two `char` values to represent, we should make sure there's a test where one of the char is one buffer and the next is in the subsequent input buffer, such that it will cause the internal encoder to do a partial encoding because it needs more data, just to make that those code paths are properly tested.

Reply:
I've added these scenarios